### PR TITLE
feat(auth): create system key

### LINKED ISSUE #4234: [auth][graphql] `createSystemAPIKey` resolver

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #4234

This introduces create system API creation mutation. It is not wored up, is intended for scaffolding a working UI.

```graphql
mutation {
  createSystemApiKey(input: { name: "system-api-key"}) {
    jwt
    apiKey {
      name
      description
      createdAt
      expiresAt
    }
  }
}
```

Payload

```json
{
  "data": {
    "createSystemApiKey": {
      "jwt": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoic3lzdGVtLWFwaS1rZXkiLCJkZXNjcmlwdGlvbiI6bnVsbCwiaWF0IjoxNzIzNjg3NTI4LCJpZCI6OH0.29TOm3JM2aY5vdALWnNlLA7cJXM1r3UZu3ApiqfEhpg",
      "apiKey": {
        "name": "system-api-key",
        "description": null,
        "createdAt": "2024-08-15T02:05:28+00:00",
        "expiresAt": null
      }
    }
  }
}
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -61,6 +61,7 @@ dependencies = [
   "arize-phoenix-evals>=0.13.1",
   "fastapi",
   "pydantic>=1.0,!=2.0.*,<3,", # exclude 2.0.* since it does not support the `json_encoders` configuration setting
+  "pyjwt",
```
Comment (mikeldking) [inline pyproject.toml:64]: @axiomofjoy not 100% sure on the authlib dependency yet and so just minimizing the dependencies for now. I think it's pretty interchangeable if needed.

Comment (axiomofjoy) [reply pyproject.toml:64]: sg

File: src/phoenix/server/app.py [hunk 2]
```diff
@@ -276,7 +276,26 @@ def create_graphql_router(
...
+        export_path (Path): the file path to export data to for download (legacy)
+        last_updated_at (CanGetLastUpdatedAt): How to get the last updated timestamp for updates.
+        event_queue (CanPutItem[DmlEvent]): The event queue for DML events.
+        corpus (Optional[Model], optional): the corpus for UMAP projection. Defaults to None.
+        cache_for_dataloaders (Optional[CacheForDataLoaders], optional): GraphQL data loaders.
+        read_only (bool, optional): Marks the app as read-only. Defaults to False.
+        secret (Optional[str], optional): The application secret for auth. Defaults to None.
+
+    Returns:
+        GraphQLRouter: _description_
```
Comment (axiomofjoy) [inline src/phoenix/server/app.py:296]: Missed this part.

File: src/phoenix/server/api/mutations/api_key_mutations.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+class CreateSystemApiKeyMutationPayload:
+    jwt: str
+    api_key: SystemApiKey
+    query: Query
+
+
+@strawberry.input
+class CreateApiKeyInput:
+    name: str
+    description: Optional[str] = UNSET
```
Comment (axiomofjoy) [inline src/phoenix/server/api/mutations/api_key_mutations.py:27]: We'll also need the expiry.

**Review by axiomofjoy — APPROVED** [review]

File: src/phoenix/server/api/mutations/api_key_mutations.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                    name=input.name,
+                    description=input.description or None,
+                )
+                .returning(models.APIKey)
+            )
+            api_key = await session.scalar(insert_stmt)
+            assert api_key is not None
+
+        encoded_jwt = create_jwt(
+            secret=info.context.get_secret(),
```
[the remark this answers is not in the document]
Comment (mikeldking) [reply src/phoenix/server/api/mutations/api_key_mutations.py:61]: I originally had it that way - I have a new permissions guard so I decided to move it down.  Felt this is the safest option

File: src/phoenix/server/api/mutations/api_key_mutations.py (hunk 3 above)
Comment (mikeldking) [reply src/phoenix/server/api/mutations/api_key_mutations.py:27]: Okay I can add that. I wasn't sure if you wanted system keys to not expire for some reason

**Pull request merged**